src/libs/ajax/User.test.ts

```diff
@@ -1,38 +1,431 @@
-    // Act
-    const result = kvArrayToObject([
-      { key: 'foo', value: 1 },
-      { key: 'bar', value: 2 },
-      { key: 'baz', value: 3 },
-    ]);
-
-    // Assert
-    expect(result).toEqual({
-      foo: 1,
-      bar: 2,
-      baz: 3,
+import {
+  CreateTerraUserProfileRequest,
+  generateAPIBodyForCreateUserProfile,
+  generateAPIBodyForUpdateUserProfile,
+  OrchestrationUpsertTerraUserProfileRequest,
+  UpdateTerraUserProfileRequest,
+} from 'src/libs/ajax/User';
+import { TerraUserProfile } from 'src/libs/state';
+
+type AjaxCommonExports = typeof import('src/libs/ajax/ajax-common');
+jest.mock(
+  'src/auth/auth',
+  (): AjaxCommonExports => ({
+    ...jest.requireActual<AjaxCommonExports>('src/libs/ajax/ajax-common'),
+    authOpts: jest.fn().mockReturnValue({ headers: { Authorization: 'testToken' } }),
+  })
+);
+
+const completeUserProfile: TerraUserProfile = {
+  firstName: 'testFirstName',
+  lastName: 'testLastName',
+  contactEmail: 'user@example.com',
+  department: 'testDepartment',
+  institute: 'testInstitute',
+  interestInTerra: 'testInterestInTerra',
+  programLocationCity: 'testCity',
+  programLocationCountry: 'testCountry',
+  programLocationState: 'testState',
+  starredWorkspaces: 'testStarredWorkspaces',
+  title: 'testTitle',
+  researchArea: 'testResearchArea',
+} as const satisfies TerraUserProfile;
+
+const minimalCreateUserProfileRequest: CreateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+};
+
+const completeCreateUserProfileRequest: CreateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+  department: completeUserProfile.department,
+  institute: completeUserProfile.institute,
+  interestInTerra: completeUserProfile.interestInTerra,
+  title: completeUserProfile.title,
+};
+
+const minimalUpdateUserProfileRequest: UpdateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+};
+
+const completeUpdateUserProfileRequest: UpdateTerraUserProfileRequest = {
+  firstName: completeUserProfile.firstName!,
+  lastName: completeUserProfile.lastName!,
+  contactEmail: completeUserProfile.contactEmail!,
+  department: completeUserProfile.department,
+  institute: completeUserProfile.institute,
+  interestInTerra: completeUserProfile.interestInTerra,
+  title: completeUserProfile.title,
+  programLocationCity: completeUserProfile.programLocationCity,
+  programLocationState: completeUserProfile.programLocationState,
+  programLocationCountry: completeUserProfile.programLocationCountry,
+  researchArea: completeUserProfile.researchArea,
+};
+
+const NA = 'N/A';
+
+// create user profile
+describe('A create user profile request', () => {
```

Comment:
This test has so many test cases. It seems like most of them could be combined into a single case. Mind if I take a stab at it?

Reply:
Sure, go ahead! The main thing I was trying to maintain is a separation of fields that are required to be in different states (e.g. things that might be undefined vs things that are specifically expected to be undefined) to help clarify the sort of confusing web of requirements

Reply:
Maintained all the tests, but utilized `it.each` in a bunch of places to clean it up. Shaved about 180 lines off!

Reply:
oh yeah wow my brain definitely didn't pick up that you could just plug in [field] to the assertion to better utilize it.each 🤦

Reply:
I mean, tbh, I only learned about it recently. The more we know!